src/main/scala/cromwell/engine/db/slick/SlickDataAccess.scala

```diff
@@ -307,28 +308,45 @@ class SlickDataAccess(databaseConfig: Config, val dataAccess: DataAccessComponen
     runTransaction(action)
   }
 
-  override def getWorkflowsByState(states: Traversable[WorkflowState]): Future[Traversable[WorkflowInfo]] = {
+  override def getWorkflow(workflowId: WorkflowId): Future[WorkflowDescriptor] = {
+    val action = for {
+      workflowExecutionResult <- dataAccess.workflowExecutionsByWorkflowExecutionUuid(workflowId.toString).result.head
```

Comment:
I'll put a comment if you want but I don't really know what the comment should be.  It probably doesn't help that I don't know these data structures super well, I'm guessing if `.head` doesn't work that maybe an exception is thrown?  Would that be captured in the `Future` as a `Future.failure` or something?  Anybody know?